src/Data/Semiring/Free.purs

```diff
@@ -19,7 +23,12 @@ runFree (Free xs) = xs
 -- | Lift a value of type `a` to a value of type `Free a`
 free :: forall a. a -> Free a
 free a = Free [[a]]
- 
+
+-- | Apply function `fn` to every element of `Free a` 
+mapFree :: forall a b. (a -> b) -> Free a -> Free b
```

Comment:
Looks good, but I think we should prefer the `Functor` instance. Could you please just move this inside there?